AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckSelectionDialog.java

```diff
@@ -182,14 +183,20 @@ public boolean onQueryTextChange(String newText) {
     }
 
     private void addDeckDialog(String parentDeckPath) {
-        int dialogTitle = parentDeckPath.isEmpty() ? R.string.new_deck :  R.string.create_subdeck;
-        CreateDeckDialog deckDialog = new CreateDeckDialog(requireActivity(), dialogTitle);
-        deckDialog.showCreateDeckDialog();
-
-        deckDialog.builder.onPositive((dialog, which) -> {
-            String deckNameWithParentName = parentDeckPath + deckDialog.getDeckName();
-            boolean success = deckDialog.createDeck(deckNameWithParentName);
-            if (success){
+        int dialogTitle;
+        if (parentDeckPath.isEmpty()) {
+            dialogTitle = R.string.new_deck;
+        } else {
+            dialogTitle = R.string.create_subdeck;
+        }
+        EditText dialogEditText = new EditText(requireActivity());
+        CreateDeckDialog createDeckDialog = new CreateDeckDialog(requireActivity(), dialogTitle , dialogEditText);
```

Comment:
See if you can move `dialogEditText` inside the constructor

```suggestion
        CreateDeckDialog createDeckDialog = new CreateDeckDialog(requireActivity(), dialogTitle, dialogEditText);
```